### PR TITLE
Fixes #17597: NPE when agent key is malformed

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17597

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/nodes/NodeInfo.scala [hunk 1]
```diff
@@ -280,7 +282,7 @@ object CFEngineKey {
    */
   protected def getCfengineDigest(pubkeyInfo: SubjectPublicKeyInfo, algo: String): Box[String] = {
     for {
-      keyFactory <- pubkeyInfo.getAlgorithm.getAlgorithm match {
+t-      keyFactory <- pubkeyInfo.getAlgorithm.getAlgorithm match {
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/nodes/NodeInfo.scala:285]: typo

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
OK, squash merging this PR

**Pull request merged**